src/data_designer/config/preview_results.py

```diff
@@ -29,4 +30,5 @@ def __init__(
         """
         self.dataset: pd.DataFrame | None = dataset
         self.analysis: DatasetProfilerResults | None = analysis
```

Comment:
A couple things

- These are all understood to be "preview" objects (e.g., the dataset is the preview dataset), so I don't think we need to have the `_preview` on the attribute name. So, maybe just `processor_artifacts` makes sense. 

- Need to update the docstring